### ISSUE TITLE
[human] Chat Agent: Auto-Generate Full GitHub Issue Metadata (Priority, Size, Labels, Dates, Branch)

### ISSUE THREAD

**Issue by Boykai** [Boykai/github-workflows#1695]
> **Parent Issue:** #1687 — Chat Agent: Auto-Generate Full GitHub Issue Metadata (Priority, Size, Labels, Dates, Branch)

## 🤖 Agent Task: `human`

This is a manual human task. Complete the work described below, then close this issue or comment 'Done!' on the parent issue to continue the pipeline.

---

## Original Request

> Update to ensure app chat agent generates all metadata for a GitHub Issue that would be interesting. For example, tags, priority, size, estimate, start date, target date, labels, development (parent branch). The possible values for these fields should be queried and stored in cache or sqlite for the AI chat agent to then select from when creating the description for the ticket. These fields should be sent when creating the GitHub Issue.

## User Story

As a developer using the app's chat agent to create GitHub Issues, I want the agent to automatically generate and populate all relevant issue metadata — including tags, priority, size, estimate, start date, target date, labels, and development/parent branch — so that issues are fully formed and actionable without manual post-creation edits. The agent should query GitHub for all possible field values, cache them locally (in SQLite or an in-memory cache), and select appropriate values at issue creation time before submitting the fully-populated issue via the GitHub API.

## UI/UX Description

When the chat agent prepares to create a GitHub Issue, it should display a structured preview or confirmation card showing all generated metadata fields side-by-side with the issue title and description. Fields include: tags (multi-select), priority (P0–P3 or equivalent), size (T-shirt or story points), time estimate, start date, target date, labels (sourced from the repo's actual label list), and development/parent branch (sourced from the repo's branch list). Each auto-populated value should be visually distinguishable (e.g., badge or chip style) and ideally editable by the user before final submission. The agent should indicate when values were resolved from cache vs. freshly fetched. On submission, all metadata fields must be included in the GitHub API payload.

## Functional Requirements

- System MUST query the GitHub API for all available repository labels, milestones, projects, and branches when initializing or refreshing metadata context, and store results in a local SQLite table or in-memory cache keyed by repository.
- System MUST use cached metadata values (labels, branches, milestones) as the selection pool when the AI chat agent generates issue metadata, ensuring no invalid or non-existent values are submitted to the GitHub API.
- System MUST generate values for all of the following fields for every GitHub Issue created via the chat agent: tags, priority, size, estimate (hours or points), start date, target date, labels (mapped to actual repo labels), and development/parent branch.
- System MUST include all generated metadata fields in the GitHub Issues API creation payload (e.g., labels array, milestone, assignees, project fields, and branch reference via development link or custom field).
- System MUST invalidate and re-fetch cached metadata (labels, branches, etc.) when the cache is stale (e.g., TTL exceeded or on explicit user refresh), and SHOULD surface a 'Refreshing metadata…' indicator in the UI.
- System SHOULD allow the user to review and override any AI-generated metadata field (priority, size, labels, dates, branch) in the chat interface before the issue is submitted to GitHub.
- System MUST persist the metadata cache in SQLite so that it survives app restarts and is available immediately on next session without requiring a full re-fetch.
- System SHOULD gracefully handle GitHub API errors during metadata fetching (e.g., rate limits, network failures) by falling back to previously cached values and notifying the user of the degraded state.
- System MUST map AI-determined priority and size values to the closest matching repo label (e.g., 'P1', 'size:M') from the cached label list before submission, rather than submitting free-text values.
- System SHOULD log each issue creation event with the full metadata payload sent to GitHub for debugging and audit purposes.

## Technical Notes

Implement a MetadataService (or equivalent module) responsible for fetching and caching GitHub repo metadata: labels (`GET /repos/{owner}/{repo}/labels`), branches (`GET /repos/{owner}/{repo}/branches`), milestones, and project fields via the GitHub GraphQL API if Projects v2 are used. Store results in a SQLite table (e.g., `github_metadata_cache`) with columns for repo, field_type, value, and fetched_at timestamp; apply a configurable TTL (e.g., 1 hour) for cache invalidation. At issue-creation time, inject the cached metadata as structured context into the AI chat agent's system prompt so it selects from real values only. For the 'development' (parent branch) field, use the GitHub Issues API `body` custom mention or the repository's branch association depending on API support; consider using the `development` field available via GitHub's GraphQL `createIssue` mutation or a linked pull request reference. Dates (start date, target date) should be stored as ISO 8601 strings and mapped to milestone due dates or custom project fields.

## Metadata

| Field | Value |
|-------|-------|
| Priority | P1 |
| Size | L |
| Estimate | 16.0h |
| Start Date | 2026-03-05 |
| Target Date | 2026-03-07 |
| Labels | `feature`, `backend`, `frontend`, `database`, `api`, `ux`, `ai-generated` |

---
*Sub-issue created for agent `human` — see parent issue #1687 for full context*

**Comment by Boykai** [comment]
**`.github/agents/copilot-instructions.md`** (generated by `human`)

---

# GitHub Workflows Chat — Development Guidelines

Last updated: 2026-03-04

> **Important:** Always use search tools (e.g., Context7 MCP, Microsoft Docs MCP, or web search) to look up the most up-to-date documentation when working with any and all libraries, frameworks, and APIs. Never rely solely on training data — verify current syntax, options, and best practices from official sources before writing or modifying code.

## Tech Stack

### Backend (Python)
- **Runtime**: Python ≥3.11 (Docker image: 3.12-slim)
- **Framework**: FastAPI ≥0.109, Uvicorn ≥0.27
- **Data validation**: Pydantic ≥2.5, pydantic-settings ≥2.1
- **HTTP client**: httpx ≥0.26
- **Database**: SQLite via aiosqlite ≥0.20 (WAL mode, migration-managed schema, file-backed at `/app/data/settings.db`)
- **AI / Agents**: github-copilot-sdk ≥0.1, agent-framework-core ≥1.0.0a1, openai ≥1.0, azure-ai-inference ≥1.0.0b1
- **Auth**: python-jose[cryptography] ≥3.3 (JWT)
- **Utilities**: tenacity ≥8.2 (retries), websockets ≥12.0, python-multipart ≥0.0.6, pyyaml ≥6.0.1
- **Linting**: ruff ≥0.9 (target py311, line-length 100)
- **Type checking**: pyright ≥1.1.400 (targets Python 3.12, basic mode)
- **Testing**: pytest ≥7.4, pytest-asyncio ≥0.23 (auto mode), pytest-cov ≥4.1

### Frontend (TypeScript)
- **Runtime**: Node 20 (Docker: node:20-alpine build → nginx:alpine production)
- **Language**: TypeScript ~5.4 (target ES2022)
- **Framework**: React ^18.3, React DOM ^18.3
- **Build**: Vite ^5.4
- **Server state**: @tanstack/react-query ^5.17
- **Styling**: Tailwind CSS ^3.4, Shadcn UI (Radix UI primitives), class-variance-authority, clsx, tailwind-merge, lucide-react (icons)
- **Drag & drop**: @dnd-kit/core ^6.3, @dnd-kit/sortable ^10.0, @dnd-kit/modifiers ^9.0
- **Real-time**: socket.io-client ^4.7
- **Linting**: ESLint ^9 (flat config, typescript-eslint, react-hooks), Prettier ^3.2
- **Testing**: Vitest ^4.0 (happy-dom), @testing-library/react ^16.3, Playwright ^1.58 (E2E, Chromium)

### Infrastructure
- **Containers**: Docker Compose — 3 services on `ghchat-network` bridge
  - `backend` (port 8000) — FastAPI app, `ghchat-data` volume for SQLite
  - `frontend` (port 5173 → nginx :80) — static SPA
  - `signal-api` (port 8080 internal) — `bbernhard/signal-cli-rest-api:latest`, JSON-RPC mode
- **CI**: GitHub Actions — backend (ruff, pyright, pytest) + frontend (eslint, tsc, vitest, vite build) + Docker image build verification
- **CI actions**: pinned to SHA with explicit `permissions: contents: read`

## Project Structure

```
backend/
  src/
    api/              # FastAPI route handlers
    middleware/        # Request/response middleware
    migrations/       # SQL migration files (001–010)
    models/           # Pydantic models (board, chat, settings, task, user, agent_creator, etc.)
    prompts/          # AI prompt templates
    services/         # Business logic
      copilot_polling/ # Background polling loop for Copilot agent pipeline
      github_projects/ # GitHub GraphQL API client
      workflow_orchestrator/ # Agent pipeline config, state, transitions
      agent_creator.py # #agent command: guided custom agent creation flow
  tests/
    unit/             # pytest unit tests
    integration/      # pytest integration tests
    helpers/          # Shared factories and mocks

frontend/
  src/
    components/       # React components (auth, board, chat, common, settings, ui)
    hooks/            # Custom React hooks (useAuth, useChat, useProjectBoard, useWorkflow, etc.)
    lib/              # Shared utilities
    pages/            # Route pages
    services/         # API client functions
    types/            # TypeScript type definitions
    utils/            # Helper functions
  e2e/                # Playwright E2E tests
```

## Commands

```bash
# Backend
cd backend && source .venv/bin/activate
ruff check src/ tests/          # lint
ruff format src/ tests/         # format
pyright src/                    # type check
pytest tests/unit/ -q           # unit tests
pytest tests/ --cov=src         # all tests with coverage

# Frontend
cd frontend
npm run lint                    # eslint
npm run type-check              # tsc --noEmit
npm test                        # vitest
npm run build                   # production build
npx playwright test             # E2E tests
```

## Code Style

- **Python**: Follow ruff defaults — double quotes, 100 char lines, isort with `known-first-party = ["src"]`
- **TypeScript**: Strict mode, ESNext modules, path alias `@/` → `src/`
- **Tests**: Arrange-Act-Assert pattern. Backend factories in `tests/helpers/factories.py`, mocks in `tests/helpers/mocks.py`. Frontend test setup in `src/test/setup.ts`
- **Commits**: Conventional commits (`feat:`, `fix:`, `docs:`, `refactor:`, `test:`, `chore:`)

## Key Architecture Notes

- Database migrations run automatically on startup (sequential SQL files in `src/migrations/`)
- SQLite uses WAL mode, busy_timeout=5000ms, foreign_keys=ON
- Copilot polling loop runs as a background task, scanning project board items every N seconds
- Sub-issues use title convention `[agent-name] Parent Title` and are filtered from the polling loop
- Agent pipeline state tracks per-issue progress through configurable status→agent mappings
- Session auth uses secure HTTP-only cookies set via a shared helper


## Active Technologies
- Python ≥3.11 (backend), TypeScript ~5.4 (frontend) + FastAPI ≥0.109, React 18.3, TanStack Query 5, Tailwind CSS 3.4, aiosqlite ≥0.20, PyYAML ≥6.0 (017-agents-section)
- SQLite via aiosqlite (`agent_configs` table exists), GitHub repo files via GraphQL API (017-agents-section)

## Recent Changes
- 017-agents-section: Added Python ≥3.11 (backend), TypeScript ~5.4 (frontend) + FastAPI ≥0.109, React 18.3, TanStack Query 5, Tailwind CSS 3.4, aiosqlite ≥0.20, PyYAML ≥6.0
